content/articles/how-is-textInputLayout-different-from-edittext/index.md

```diff
@@ -0,0 +1,304 @@
+Like to filled text field and outlined text filled, theming text field has two height variants, that is, it consists of standard and dense text fields.
+
+### Adding TextInputLayout to our android app.
+Now, I am going to create a new project to show you how to work with TextInputLayout in an android application.I will be using a single activity application and we will do  all cool stuff inside activity_main.
+
+### Step 1:
+Creating Project
+First you have to open a new project in your android studio. Then open your build.gradle . Add material dependency and click sync now.
+
+### Step 2: Adding material dependency
+```gradle 
+//Material  dependency 
+implementation 'com.google.android.material:material:1.4.0'
+``` 
+
+### Step 3: Adding TextInputLayout to your activity_main file
+Then we reach to the activity_main file and delete the default TextView add TextInputLayout as a child layout inside ConstraintLayout and constraint it and give it some margin.
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".MainActivity">
+    <com.google.android.material.textfield.TextInputLayout
+        android:layout_width="409dp"
+        android:layout_height="wrap_content"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:layout_editor_absoluteX="1dp"
+        tools:ignore="MissingConstraints">
+    </com.google.android.material.textfield.TextInputLayout>
+</androidx.constraintlayout.widget.ConstraintLayout>
+
+```
+### Step 4: Adding TextInputEditText into TextInputLayout
+Inside TextInputLayout create TextInputEditText, where user can input or edit the data.
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".MainActivity">
+    <com.google.android.material.textfield.TextInputLayout
+        android:layout_width="409dp"
+        android:layout_height="wrap_content"
+        app:layout_constraintBottom_toBottomOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        tools:layout_editor_absoluteX="1dp">
+        <com.google.android.material.textfield.TextInputEditText
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"/>
+    </com.google.android.material.textfield.TextInputLayout>
+</androidx.constraintlayout.widget.ConstraintLayout>
```

Comment:
Ensure that your code is indented correctly. Make the same change to all code snippets.